src/helm/benchmark/presentation/summarize.py

```diff
@@ -315,6 +319,133 @@ def read_runs(self):
                 self.group_adapter_to_runs[group_name][adapter_spec].append(run)
                 self.group_scenario_adapter_to_runs[group_name][scenario_spec][adapter_spec].append(run)
 
+    def read_overlap_stats(self):
+        """
+        Load the overlap stats in the run suite path.
+        Concretely:
+            - get group -> scenario_spec information from self.runs
+                run_spec data
+            - read the files in the data_overlap directory in run_suit_path
+                which are scenario_spec -> overlap ids
+            - get aggregate stats for group -> overlap ratio
+        """
+
+        def get_group_to_scenario_specs(run_specs: List[RunSpec]) -> Dict[RunSpec, ScenarioSpec]:
```

Comment:
Would it? I'd rather have a few nested functions that abstract this logic out rather than a huge function that needs to be read line by line, especially when read using an editor which can jump to the functions. It'll also be easier to share this logic if the need arises as is